cmd/apiregister-gen/generators/versioned_generator.go

```diff
@@ -57,14 +57,26 @@ func (d *versionedGenerator) Finalize(context *generator.Context, w io.Writer) e
 var VersionedAPITemplate = `
 var (
 	{{ range $api := .Resources -}}
-	{{$api.Group}}{{$api.Kind}}Storage = builders.NewApiResource( //  Resource endpoint
-			{{ $api.Group }}.Internal{{ $api.Kind }},
+
+	{{if $api.REST }}
+		{{$api.Group}}{{$api.Kind}}Storage = builders.NewApiResourceWithStorage( // Resource status endpoint
+			{{ $api.Group }}.Internal{{ $api.Kind }}Status,
```

Comment:
> If the intention is to store the object, why aren't we using the existing storage impl? Are you trying to override some piece of the behavior?

I want to store the objects differently than what today's REST/storage layer does.